buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild/propagated-env-variables.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild
+
+import org.gradle.api.tasks.testing.Test
+
+
+val propagatedEnvAllowList = listOf(
+    // Obviously necessary
+    "HOME",
+    "SHELL",
+    "TERM",
+
+    // Otherwise Windows complains "Unrecognized Windows Sockets error: 10106"
+    "SystemRoot",
+    "OS",
+
+    // For Android tests
+    "ANDROID_HOME",
+
+    // Used by Visual Studio
+    "USERNAME",
+    "USER",
+    "USERDOMAIN",
+    "USERPROFILE",
+    "LOCALAPPDATA",
+
+    // Used by Gradle test infrastructure
+    "REPO_MIRROR_URL",
+
+    // temp dir
+    "TMPDIR",
+    "TMP",
+    "TEMP",
+
+    // Seems important on Windows
+    "ALLUSERSPROFILE",
+    "PUBLIC",
+    "windir"
+)
+
+
+val propagatedBlockList = listOf(
+    "api_key",
+    "access_key",
+    "apikey",
+    "accesskey",
+    "password",
+    "token",
+    "credential",
+    "auth"
+)
+
+
+fun Test.configurePropagatedEnvVariables() {
+    environment = System.getenv().entries.mapNotNull(::sanitize).toMap()
+    environment.forEach { (key, value) ->
+        if (propagatedBlockList.any { key.contains(it, true) } || propagatedBlockList.any { value.toString().contains(it, true) }) {
```

Comment:
I am not sure why we check the value of the environment variable here against the block list. Which block list entry would be part of the value which indicates that we pass a secret?

Reply:
This is necessary. After I wrote this line, I immediately fond a leaked password on my local machine: `TEAMCITY_ACCESS_TOKEN`. Without this check, I could have uploaded this token to public GE build scan.

Reply:
This might also cause branch name `blindpirate/fix-leaked-token` to be rejected by TeamCity build.

Reply:
I think the first part of this check is really good (`propagatedBlockList.any { key.contains(it, true) }`), I was wondering about the second part: `propagatedBlockList.any { value.toString().contains(it, true) })`. Why is it a problem if the value of an environment variable contains things from the `propagatedBlockList`?

Reply:
OK I see your point. Probably not.